# Trim Lab Data
The objective of this notebook is to trim *kidney_disease_random_1000_patients.csv.gz* and *normal_kidney_patients.csv.gz* by reducing the types of labs included to those that are specified in *Minimal Reducedlabitems - Reducedlabitems.csv*. This yields a smaller data set that still contains all of the information that is relevant to our study. We start by trimming *kidney_disease_random_1000_patients.csv.gz*.

In [2]:
import pandas as pd
import numpy as np
import json

In [21]:
kidney_disease_data = pd.read_csv('../Data/kidney_disease_random_1000_patients.csv.gz',
                                 compression = 'gzip')
kidney_disease_data.head()

,labevent_id,subject_id,hadm_id,specimen_id,itemid,order_provider_id,charttime,storetime,value,valuenum,valueuom,ref_range_lower,ref_range_upper,flag,priority,comments
0,917899,10061124,24090860.0,19944894,50868,NaN,2168-08-03 19:30:00,2168-08-03 20:03:00,11,11.0,mEq/L,8.0,20.0,NaN,STAT,NaN
1,917900,10061124,24090860.0,19944894,50882,NaN,2168-08-03 19:30:00,2168-08-03 20:03:00,28,28.0,mEq/L,22.0,32.0,NaN,STAT,NaN
2,917901,10061124,24090860.0,19944894,50902,NaN,2168-08-03 19:30:00,2168-08-03 20:03:00,106,106.0,mEq/L,96.0,108.0,NaN,STAT,NaN
3,917902,10061124,24090860.0,19944894,50912,NaN,2168-08-03 19:30:00,2168-08-03 20:03:00,1.3,1.3,mg/dL,0.5,1.2,abnormal,STAT,NaN
4,917903,10061124,24090860.0,19944894,50920,NaN,2168-08-03 19:30:00,2168-08-03 20:03:00,NaN,NaN,NaN,NaN,NaN,NaN,STAT,"Using this patient's age, gender, and serum cr..."


In [3]:
reduced_labs = pd.read_csv('../Data/Minimal Reducedlabitems - Reducedlabitems.csv')
reduced_labs.head()

,itemid,label,fluid,category,Unnamed: 4
0,50853,25-OH Vitamin D,Blood,Chemistry,NaN
1,50855,Absolute Hemoglobin,Blood,Chemistry,NaN
2,50861,Alanine Aminotransferase (ALT),Blood,Chemistry,NaN
3,50862,Albumin,Blood,Chemistry,NaN
4,50863,Alkaline Phosphatase,Blood,Chemistry,NaN


In [5]:
reduced_labs['itemid'].shape[0]

105

In [23]:
reduced_ids = list(reduced_labs['itemid'])
labevents_cols = ['labevent_id',
                 'subject_id',
                 'hadm_id',
                 'specimen_id',
                 'itemid',
                 'order_provider_id',
                 'charttime',
                 'storetime',
                 'value',
                 'valuenum',
                 'valueuom',
                 'ref_range_lower',
                 'ref_range_upper',
                 'flag',
                 'priority',
                 'comments']
trimmed_kidney_disease_data = pd.DataFrame(columns = labevents_cols)
for lab in reduced_ids:
    lab_df = kidney_disease_data.loc[kidney_disease_data['itemid'] == lab]
    trimmed_kidney_disease_data = pd.concat([trimmed_kidney_disease_data, lab_df])

C:\Users\steven\AppData\Local\Temp\ipykernel_5508\833474255.py:21: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  trimmed_kidney_disease_data = pd.concat([trimmed_kidney_disease_data, lab_df])


In [24]:
trimmed_kidney_disease_data.to_csv('../Data/trimmed_kidney_disease_random_1000_patients.csv.gz',
                                  compression = 'gzip',
                                  index = False)
print('The number of lab items in the original data set is', kidney_disease_data['itemid'].value_counts().shape[0], '\n')
print('The number of lab items in the new data set is', trimmed_kidney_disease_data['itemid'].value_counts().shape[0])

The number of lab items in the original data set is 826 

The number of lab items in the new data set is 95


As may be seen from the output of the cell above, the original data set included 826 lab items, while the trimmed data set only includes 95. The remainder of this notebook trims *normal_kidney_patients.csv.gz* in the same manner. 

In [25]:
normal_kidney_data = pd.read_csv('../Data/normal_kidney_patients.csv.gz',
                                compression = 'gzip')
normal_kidney_data.head()

,labevent_id,subject_id,hadm_id,specimen_id,itemid,order_provider_id,charttime,storetime,value,valuenum,valueuom,ref_range_lower,ref_range_upper,flag,priority,comments
0,617746,10039302,25044971.0,12440596,50868,NaN,2118-02-06 00:10:00,2118-02-06 01:28:00,16,16.0,mEq/L,8.0,20.0,NaN,STAT,NaN
1,617747,10039302,25044971.0,12440596,50882,NaN,2118-02-06 00:10:00,2118-02-06 01:28:00,26,26.0,mEq/L,22.0,32.0,NaN,STAT,NaN
2,617748,10039302,25044971.0,12440596,50893,NaN,2118-02-06 00:10:00,2118-02-06 01:28:00,9.6,9.6,mg/dL,8.4,10.2,NaN,STAT,NaN
3,617749,10039302,25044971.0,12440596,50902,NaN,2118-02-06 00:10:00,2118-02-06 01:28:00,106,106.0,mEq/L,96.0,108.0,NaN,STAT,NaN
4,617750,10039302,25044971.0,12440596,50912,NaN,2118-02-06 00:10:00,2118-02-06 01:28:00,1.0,1.0,mg/dL,0.4,1.1,NaN,STAT,NaN


In [26]:
trimmed_normal_kidney_data = pd.DataFrame(columns = labevents_cols)
for lab in reduced_ids:
    lab_df = normal_kidney_data.loc[normal_kidney_data['itemid'] == lab]
    trimmed_normal_kidney_data = pd.concat([trimmed_normal_kidney_data, lab_df])

C:\Users\steven\AppData\Local\Temp\ipykernel_5508\927813807.py:4: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  trimmed_normal_kidney_data = pd.concat([trimmed_normal_kidney_data, lab_df])


In [27]:
trimmed_normal_kidney_data.to_csv('../Data/trimmed_normal_kidney_patients.csv.gz',
                                 compression = 'gzip',
                                 index = False)
print('The number of lab items in the original data set is', normal_kidney_data['itemid'].value_counts().shape[0], '\n')
print('The number of lab items in the new data set is', trimmed_normal_kidney_data['itemid'].value_counts().shape[0])

The number of lab items in the original data set is 693 

The number of lab items in the new data set is 76


Based on the output above, *normal_kidney_patients.csv.gz* has been trimmed from $693$ lab items to $76$ lab items.